azure_provisioning_e2e/tests/test_async_symmetric_enrollments.py

```diff
@@ -0,0 +1,110 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import time
+from azure_provisioning_e2e.iothubservice20180630.service_helper import Helper
+from azure.iot.device.aio import ProvisioningDeviceClient
+from provisioningserviceclient import ProvisioningServiceClient, IndividualEnrollment
+from provisioningserviceclient.protocol.models import AttestationMechanism, ReprovisionPolicy
+import pytest
+import logging
+import os
+
+
+pytestmark = pytest.mark.asyncio
+logging.basicConfig(level=logging.DEBUG)
+
+
+PROVISIONING_HOST = os.getenv("PROVISIONING_DEVICE_ENDPOINT")
+ID_SCOPE = os.getenv("PROVISIONING_DEVICE_IDSCOPE")
+conn_str = os.getenv("PROVISIONING_SERVICE_CONNECTION_STRING")
+service_client = ProvisioningServiceClient.create_from_connection_string(
+    os.getenv("PROVISIONING_SERVICE_CONNECTION_STRING")
+)
+service_client = ProvisioningServiceClient.create_from_connection_string(conn_str)
+device_registry_helper = Helper(os.getenv("IOTHUB_CONNECTION_STRING"))
+
+
+@pytest.mark.it(
+    "A device gets provisioned to the linked IoTHub with the device_id equal to the registration_id of the individual enrollment that has been created with a symmetric key authentication"
+)
+async def test_device_register_with_no_device_id_for_a_symmetric_key_individual_enrollment():
+    try:
+        individual_enrollment_record = create_individual_enrollment("e2e-dps-legilimens")
+
+        registration_id = individual_enrollment_record.registration_id
+        symmetric_key = individual_enrollment_record.attestation.symmetric_key.primary_key
+
+        await result_from_register(registration_id, symmetric_key)
```

Comment:
Personally, I find that these factored out snippets of code in general hurt the readability of tests - I prefer them longer and explicit, even if that creates repetition. But, for E2E it doesn't matter as much as unittests, since we aren't constantly modifying them, so I don't care too much one way or the other.